code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm

```diff
@@ -467,3 +467,551 @@ GLOBAL_LIST_INIT(acid_spray_hit, typecacheof(list(/obj/structure/barricade, /obj
+	ability_cost = 50
+	cooldown_duration = 12 SECONDS
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_ITEM_THROW,
+	)
+	/// If they have moved at least a single tile since picking up an item. Used as a single-tile grace period before dropping.
+	var/single_tile_grace_period = TRUE
+	/// If we are holding an item.
+	var/obj/item/held_item
+	/// Mutable appearance of the held item.
+	var/mutable_appearance/held_appearance
+
+/datum/action/ability/activable/xeno/item_throw/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return FALSE
+	// Any target is acceptable if we have an item.
+	if(held_item)
+		return TRUE
+	// If we do not, we only want items.
+	if(!isitem(A) || isgrabitem(A))
+		if(!silent)
+			A.balloon_alert(owner, "not an item")
+		return FALSE
+	var/obj/item/item_atom = A
+	if(item_atom.anchored)
+		if(!silent)
+			item_atom.balloon_alert(owner, "item is anchored")
+		return FALSE
+	if(!owner.Adjacent(item_atom))
+		if(!silent)
+			item_atom.balloon_alert(owner, "too far")
+		return FALSE
+
+/datum/action/ability/activable/xeno/item_throw/use_ability(atom/A)
+	if(!held_item)
+		// Grab the floor item like a rouny.
+		playsound(owner, 'sound/voice/alien/pounce2.ogg', 30)
+		var/obj/item/interacted_item = A
+		interacted_item.forceMove(owner)
+		held_item = interacted_item
+		held_appearance = mutable_appearance(interacted_item.icon, interacted_item.icon_state)
+		held_appearance.layer = ABOVE_OBJ_LAYER
+		RegisterSignal(owner, COMSIG_MOVABLE_MOVED, PROC_REF(on_move))
+		RegisterSignal(owner, COMSIG_ATOM_DIR_CHANGE, PROC_REF(owner_turned))
+		owner.add_movespeed_modifier(MOVESPEED_ID_OPPRESSOR_ITEM_GRAB, TRUE, 0, NONE, TRUE, 2)
+		owner_turned(null, null, owner.dir)
+		succeed_activate()
+		ability_cost = 0 // Throwing will cost nothing (and won't cause a warning if not enough plasma).
+		single_tile_grace_period = TRUE
+		return
+	// Throw!
+	owner.remove_movespeed_modifier(MOVESPEED_ID_OPPRESSOR_ITEM_GRAB)
+	held_item.throwforce += held_item.w_class * 15 // Ranges from 15 - 90 damage.
+	RegisterSignal(held_item, COMSIG_MOVABLE_POST_THROW, PROC_REF(on_throwend))
+	held_item.forceMove(get_turf(owner))
+	// Speed 5 for maximum damage.
+	held_item.throw_at(A, max(2, 11 - (held_item.w_class * 2)), 5) // Ranges from 9 - 2 tiles.
+	held_item = null
+	owner.overlays -= held_appearance
+	playsound(xeno_owner, 'sound/effects/throw.ogg', 30, 1)
+	succeed_activate()
+	ability_cost = initial(ability_cost)
+	add_cooldown()
+	UnregisterSignal(owner, COMSIG_MOVABLE_MOVED)
+	UnregisterSignal(owner, COMSIG_ATOM_DIR_CHANGE)
+
+/datum/action/ability/activable/xeno/item_throw/proc/on_throwend(datum/source)
+	SIGNAL_HANDLER
+	UnregisterSignal(source, COMSIG_MOVABLE_POST_THROW)
+	var/obj/item/item_source = source
+	item_source.throwforce -= item_source.w_class * 15
+
+/// On move, drop the item if grace period is over. Otherwise, remove the grace period.
```

Comment:
grace period isnt really accurate to what this is doing. Its a movement check, there's no time factor.